airbyte-integrations/connectors/source-shopify/source_shopify/shopify_graphql/bulk/query.py

```diff
@@ -0,0 +1,1387 @@
+    }
+
+    def process_fulfillment_order(self, record: Mapping[str, Any], shop_id: Optional[int] = 0) -> Mapping[str, Any]:
+        # addings
+        record["shop_id"] = shop_id
+        record["order_id"] = record.get(BULK_PARENT_KEY)
+        # unnest nested locationId to the `assignedLocation`
+        location_id = record.get("assignedLocation", {}).get("location", {}).get("locationId")
+        record["assignedLocation"]["locationId"] = location_id
+        record["assigned_location_id"] = location_id
+        # create nested placeholders for other parts
+        record["line_items"] = []
+        record["merchant_requests"] = []
+        # cleaning
+        record.pop(BULK_PARENT_KEY)
+        record.get("assignedLocation").pop("location", None)
+        # resolve ids from `str` to `int`
+        # location id
+        location = record.get("assignedLocation", {})
+        if location:
+            location_id = location.get("locationId")
+            if location_id:
+                record["assignedLocation"]["locationId"] = self.tools.resolve_str_id(location_id)
+        # assigned_location_id
+        record["assigned_location_id"] = self.tools.resolve_str_id(record.get("assigned_location_id"))
+        # destination id
+        destination = record.get("destination", {})
+        if destination:
+            destination_id = destination.get("id")
+            if destination_id:
+                record["destination"]["id"] = self.tools.resolve_str_id(destination_id)
+        # delivery method id
+        delivery_method = record.get("deliveryMethod", {})
+        if delivery_method:
+            delivery_method_id = delivery_method.get("id")
+            if delivery_method_id:
+                record["deliveryMethod"]["id"] = self.tools.resolve_str_id(delivery_method_id)
+        # order id
+        record["order_id"] = self.tools.resolve_str_id(record.get("order_id"))
+        # field names to snake for nested objects
+        # `assignedLocation`(object) field names to snake case
+        record["assignedLocation"] = self.tools.fields_names_to_snake_case(record.get("assignedLocation"))
+        # `deliveryMethod`(object) field names to snake case
+        record["deliveryMethod"] = self.tools.fields_names_to_snake_case(record.get("deliveryMethod"))
+        # `destination`(object) field names to snake case
+        record["destination"] = self.tools.fields_names_to_snake_case(record.get("destination"))
+        # `fulfillmentHolds`(list[object]) field names to snake case
+        record["fulfillment_holds"] = [self.tools.fields_names_to_snake_case(el) for el in record.get("fulfillment_holds", [])]
+        # `supportedActions`(list[object]) field names to snake case
+        record["supported_actions"] = [self.tools.fields_names_to_snake_case(el) for el in record.get("supported_actions", [])]
+        return record
+
+    def process_line_item(self, record: Mapping[str, Any], shop_id: Optional[int] = 0) -> Mapping[str, Any]:
```

Comment:
do we ever expect shop_id not to be defined?

Reply:
No.

Reply:
corrected.